File: Java/LeetCode/Greatest English Letter in Upper and Lower Case.java
// https://leetcode.com/problems/greatest-english-letter-in-upper-and-lower-case/

public String greatestLetter(String s) {
    Set<Character> uniqueChars = new HashSet<>();

    for (int i = 0; i < s.length(); ++i) {
        uniqueChars.add(s.charAt(i));
    }

    for (char c = 'Z'; c >= 'A'; --c) {
        if (uniqueChars.contains(c) && uniqueChars.contains(Character.toLowerCase(c))) {
            return "" + c;
        }
    }

    return "";
}

File: Kotlin/LeetCode/Greatest English Letter in Upper and Lower Case.kt
// https://leetcode.com/problems/greatest-english-letter-in-upper-and-lower-case/

class Solution {

    fun greatestLetter(s: String): String {
        val uniqueChars = mutableSetOf<Char>();

        s.forEach { uniqueChars.add(it) }

        for (c in 'Z' downTo 'A') {
            if (uniqueChars.contains(c) && uniqueChars.contains(c.lowercaseChar())) {
                return c.toString();
            }
        }

        return "" 
    }

}
